### PR TITLE
feat: adjust the TablePagination font color based on the background color

### PR DESCRIPTION
**Pull request by LiKang6688**
the font color in TablePagination section should adapt to the background color
Screenshot 2022-02-22 at 00 32 07

Screenshot 2022-02-17 at 10 46 08

### CODE REVIEW

File: src/mui-setup.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { createTheme } from '@mui/material/styles';
 import * as muiConfig from './mui-config.json';
 
-export default function muiSetup(direction) {
+export default function muiSetup(direction, theme) {
```
Comment (haxxmaxx) [inline src/mui-setup.js:4]: I don't know about this. We don't want this to be dynamic

Comment (LiKang6688) [reply src/mui-setup.js:4]: No, we don't. Just found that it is convenient to write like this. But gonna write those into Component.

Comment (LiKang6688) [reply src/mui-setup.js]: Updated

File: src/index.js [hunk 2]
```diff
@@ -39,9 +40,12 @@ export default function supernova(env) {
       const constraints = useConstraints();
       const translator = useTranslator();
       const selectionsAPI = useSelections();
-      const tableTheme = muiSetup(direction);
-      // the style which is from them can be found in nebula.config.js -> serve -> themes
-      const theme = useTheme();
+      const theme = {
+        ...useTheme(),
```
Comment (haxxmaxx) [inline src/index.js:44]: would this update properly?

File: src/index.js [hunk 3]
```diff
@@ -39,9 +40,12 @@ export default function supernova(env) {
...
+        backgroundColor: useTheme().getStyle('object', 'straightTable', 'backgroundColor'),
```
Comment (haxxmaxx) [inline src/index.js:45]: So this would override if you have a general backgroundColor or?

**Comment by haxxmaxx** [comment]
The light gray outline looks weird at the bottom on the green screen shot

File: src/index.js (hunk 3 above)
Comment (LiKang6688) [reply src/index.js:45]: so here what it does is just add two new properties (backgroundColor, isBackgroundDarkColor) into the object from useTheme(). It does not override anything.

**Comment by LiKang6688** [comment]
> Good find. Let me see what happens.

File: src/index.js (hunk 2 above)
Comment (LiKang6688) [reply src/index.js:44]: I think so

Comment (haxxmaxx) [reply src/index.js:44]: For consistency and robustness I'd change `theme` after you run useTheme

```suggestion
const theme = useTheme(),
theme.backgroundColor = ....
```

File: src/index.js (hunk 3 above)
Comment (haxxmaxx) [reply src/index.js:45]: ahhh, yeah there is no general backgroundColor, this is a new prop 👍

File: src/index.js (hunk 2 above)
Comment (LiKang6688) [reply src/index.js:44]: Yeah, sure

**Comment by LiKang6688** [comment]
> The light gray outline looks weird at the bottom on the green screen shot

I realized that the border color should also adapt to the background color.

**Comment by LiKang6688** [comment]
> > The light gray outline looks weird at the bottom on the green screen shot
> 
> The issue was fixed. But I realized that the border color should also adapt to the background color.

File: src/table/components/TableWrapper.jsx [hunk 4]
```diff
@@ -113,33 +113,51 @@ export default function TableWrapper(props) {
...
+      'aria-label': translator.get('SNTable.Pagination.RowsPerPage'),
+      'data-testid': 'select',
+      tabIndex: !keyboard.enabled || keyboard.active ? 0 : -1,
+    },
+    native: true,
+  };
+
   const paperStyle = {
-    border: '1px solid #D9D9D9',
+    border: theme.isBackgroundDarkColor ? '1px solid #F2F2F3' : '1px solid #D9D9D9',
```
Comment (haxxmaxx) [inline src/table/components/TableWrapper.jsx:126]: Do we do this for the old table or some other chart when we change to a dark color?

Comment (LiKang6688) [reply src/table/components/TableWrapper.jsx:126]: Screenshot 2022-02-22 at 14 24 49
Yes.

The bottom border is tricky, now it is not possible for me to let it be good in both sense and mashup. I am still finding a way.

File: src/table/components/TableWrapper.jsx [hunk 5]
```diff
@@ -113,33 +113,51 @@ export default function TableWrapper(props) {
...
+  const SelectProps = {
+    inputProps: {
+      'aria-label': translator.get('SNTable.Pagination.RowsPerPage'),
+      'data-testid': 'select',
+      tabIndex: !keyboard.enabled || keyboard.active ? 0 : -1,
+    },
+    native: true,
+  };
+
   const paperStyle = {
```
Comment (haxxmaxx) [inline src/table/components/TableWrapper.jsx:125]: I think all style objects should be defined in styling-utils. `getTableStyle` or something that returns an object with all the diffrent styles which you the destruct. As for the other files, create a useMemo that depends on theme and constraints. Would slim down this file, and make it easier to test

Comment (LiKang6688) [reply src/table/components/TableWrapper.jsx:125]: Yup, I will add it to the issue and for future refactoring.

**Review by haxxmaxx — APPROVED** [review]

File: src/table/components/TableWrapper.jsx [hunk 6]
```diff
@@ -113,33 +113,51 @@ export default function TableWrapper(props) {
...
     display: 'flex',
     justifyContent: 'flex-end',
     alignItems: 'center',
     paddingRight: 1,
-    backgroundColor: tableBackgroundColor,
+    backgroundColor: theme.backgroundColor,
     boxShadow: 'none',
     borderRadius: 0,
-    borderBottom: '1px solid #D9D9D9',
+    borderBottom: theme.isBackgroundDarkColor ? '1px solid #F2F2F3' : '1px solid #D9D9D9',
```
[the remark this answers is not in the document]
Comment (LiKang6688) [reply src/table/components/TableWrapper.jsx:150]: from straight table
Screenshot 2022-02-28 at 10 27 06

[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/table/components/TableWrapper.jsx:150]: yes it says `#F2F2F2`, not `#F2F2F3`

[the remark this answers is not in the document]
Comment (LiKang6688) [reply src/table/components/TableWrapper.jsx:150]: aha, odd. Let me check why I got #F2F2F3

[the remark this answers is not in the document]
Comment (LiKang6688) [reply src/table/components/TableWrapper.jsx]: Fixed

**Pull request merged**